### PR TITLE
Allow decreasing the output value in RBF transactions

### PR DESCRIPTION
**Pull request by andrewkozlik**
Implements https://github.com/trezor/trezor-firmware/issues/1491.

### CODE REVIEW

File: core/src/trezor/ui/layouts/tt.py [hunk 1]
```diff
@@ -389,13 +390,43 @@ def confirm_replacement(
...
+    page1 = Text("Modify amount", ui.ICON_SEND, ui.GREEN)
+    page1.normal("Address:")
+    page1.br_half()
+    page1.mono(*_split_address(address))
+
+    page2 = Text("Modify amount", ui.ICON_SEND, ui.GREEN)
+    if sign < 0:
+        page2.normal("Decrease amount by:")
+    else:
+        page2.normal("Increase amount by:")
```
Comment (andrewkozlik) [inline core/src/trezor/ui/layouts/tt.py:409]: Note that this branch is unreachable, so as an alternative  we could raise an exception here or rename the function to `confirm_decrease_output()`.

File: legacy/firmware/layout2.c [hunk 2]
```diff
@@ -502,6 +502,37 @@ void layoutConfirmReplacement(const char *description, uint8_t txid[32]) {
...
+                               int page) {
+  if (page == 0) {
+    render_address_dialog(coin, out->address, _("Modify amount for"),
+                          _("address:"), NULL);
+  } else {
+    char *question = NULL;
+    uint64_t amount_change = 0;
+    if (orig_out->amount < out->amount) {
+      question = _("Increase amount by:");
+      amount_change = out->amount - orig_out->amount;
```
Comment (andrewkozlik) [inline legacy/firmware/layout2.c:516]: Same as in the Python implementation, this branch is currently unreachable, so as an alternative we could rename the function to `layoutConfirmDecreaseOutput()`.

**Comment by andrewkozlik** [comment]
As usual, I would like two pairs of eyes to review each of the implementations for T1 and TT. I expect @matejcik and @onvej-sl to do the review, since they are already familiar with the topic, but if @prusnak would like to review one or both implementations instead of @matejcik or @onvej-sl , then feel free to do so. Since T1 doesn't support external inputs, the implementation there is a bit simpler. For more information on the TT implementation see https://github.com/trezor/trezor-firmware/issues/1491#issuecomment-785895490.

**Review by onvej-sl — APPROVED** [review]
I reviewed both core and legacy.

File: core/src/trezor/ui/layouts/tt.py (hunk 1 above)
Comment (matejcik) [reply core/src/trezor/ui/layouts/tt.py:409]: i'd keep this as-is, this is rather far away from the code that knows about the unreachability

**Review by matejcik — COMMENTED** [review]
minor comment request, otherwise LGTM

File: core/src/apps/bitcoin/sign_tx/approvers.py [hunk 3]
```diff
@@ -44,6 +44,9 @@ def __init__(self, tx: SignTx, coin: CoinInfo) -> None:
 
         self.amount_unit = tx.amount_unit
 
+    def is_payjoin(self) -> bool:
+        return self.external_in != self.orig_external_in
```
[the remark this answers is not in the document]
Comment (andrewkozlik) [reply core/src/apps/bitcoin/sign_tx/approvers.py:51]: https://github.com/trezor/trezor-firmware/pull/1500/commits/21bfdc993dd1e4f7dcc332303d604f461bdff11e

**Review by matejcik — APPROVED** [review]

**Pull request merged**